MekHQ/src/mekhq/campaign/market/contractMarket/ContractTerms.java

```diff
@@ -0,0 +1,290 @@
+package mekhq.campaign.market.contractMarket;
+
+import mekhq.campaign.mission.enums.AtBContractType;
+import mekhq.campaign.mission.enums.ContractCommandRights;
+import mekhq.campaign.universe.Faction;
+
+import java.time.LocalDate;
+
+/**
+ * Structure that resolves and stores modifiers and contract terms as shown in the Master Contract
+ * Terms Table on page 42 of CamOps (4th printing).
+ */
+public class ContractTerms {
+    private double operationsTempoMultiplier;
+    private int baseLength;
+    private double employmentMultiplier;
+    private int commandModifier;
+    private int salvageModifier;
+    private int supportModifier;
+    private int transportModifier;
+
+    public ContractTerms(AtBContractType mission, Faction employer, int reputationFactor, LocalDate date) {
+        operationsTempoMultiplier = mission.getOperationsTempoMultiplier();
+        baseLength = mission.getConstantLength();
+        addMissionTypeModifiers(mission);
+        addEmployerModifiers(employer, date);
+        addUnitReputationModifiers(reputationFactor);
+    }
+
+    public double getOperationsTempoMultiplier() {
+        return operationsTempoMultiplier;
+    }
+
+    public int getBaseLength() {
+        return baseLength;
+    }
+
+    public double getEmploymentMultiplier() {
+        return employmentMultiplier;
+    }
+
+    public ContractCommandRights getCommandRights(int roll) {
+        roll += commandModifier;
+        ContractCommandRights commandRights;
+        if (roll > 2 && roll < 8) {
+            commandRights = ContractCommandRights.HOUSE;
+        } else if (roll < 12) {
+            commandRights = ContractCommandRights.LIAISON;
+        } else {
+            commandRights = ContractCommandRights.INDEPENDENT;
+        }
+        return commandRights;
+    }
+
+    public boolean isSalvageExchange(int roll) {
+        roll += salvageModifier;
+        return roll == 2 || roll == 3;
+    }
+
+    public int getSalvagePercentage(int roll) {
+        roll += salvageModifier;
+        if (roll < 4) {
+            return 0;
```

Comment:
This would probably be better as a switch, combined with MathUtility.clamp to put roll within usable bounds without changing effective functionality.